File: src/frontend/types/album.ts
export interface VideoSegment {
  start: number;
  end: number;
}

export interface Video {
  jobId?: string;
  filename?: string;
  videoUrl: string;
  segments: VideoSegment[];
  maxQuality: string;
}

export interface SongFingerprint {
  jobId?: string;
  filename?: string;
  fingerprintId?: string;
}

export interface Song {
  id: string;
  name: string;
  artist?: string;
  position: number;
  duration: number;
  songFingerprint?: SongFingerprint;
  songImageFilename?: string;
  songLyricsFilename?: string;
  songVideo?: Video;
  createdAt?: string;
  updatedAt?: string;
}

export interface Album {
  id: string;
  albumType: AlbumType;
  title: string;
  artist: string;
  albumImageFilename?: string;
  songs: Song[];
  createdAt?: string;
  updatedAt?: string;
}

export const albumTypes = ['CD', 'LP', 'Casette'] as const;
export type AlbumType = (typeof albumTypes)[number];

export type MetadataFileType =
  | 'AlbumImage'
  | 'SongImage'
  | 'Song'
  | 'SongLyrics'
  | 'SongVideo';

File: src/backend/VisualAudio.Data/Albums/Models/Album.cs
namespace VisualAudio.Data.Albums.Models
{
    public class Album
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public AlbumType AlbumType { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Artist { get; set; } = string.Empty;
        public string? AlbumImageFilename { get; set; }
        public IEnumerable<Song> Songs { get; set; } = [];
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; } = DateTime.UtcNow;
    }

    public enum AlbumType
    {
        CD, LP, Casette
    }
}
